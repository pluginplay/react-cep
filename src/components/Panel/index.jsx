import React from 'react'
import styled from 'styled-components'

class Panel extends React.Component {
  render () {
    return (
      <div id={'panel'} className={this.props.className}>
        {this.props.children}
      </div>
    )
  }
}

export default styled(Panel)`
  background: ${props => props.theme.ae.panelBackgroundColor};
  color: white;
  font-family: "${props => props.theme.ae.baseFontFamily}", "Segoe UI", "San Francisco", sans-serif;
  font-size: ${props => props.theme.ae.baseFontSize}pt;
  position: relative;
`
