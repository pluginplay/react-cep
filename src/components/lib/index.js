import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
