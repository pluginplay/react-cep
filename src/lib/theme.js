import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
