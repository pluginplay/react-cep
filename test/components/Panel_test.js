import { getDefaultTheme } from '../../src/lib/theme'
import React from 'react'
import Panel from '../../src/components/Panel'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<Panel theme={getDefaultTheme()}><div>Foo Bar</div></Panel>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
