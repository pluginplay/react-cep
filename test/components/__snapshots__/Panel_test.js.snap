// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa bytmHV"
  id="panel"
>
  <div>
    Foo Bar
  </div>
</div>
`;
